eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -463,10 +466,21 @@ object PeerConnection {
 
   case class DelayedRebroadcast(rebroadcast: Rebroadcast)
 
-  sealed trait BadMessage
-  case class InvalidSignature(r: RoutingMessage) extends BadMessage
-  case class InvalidAnnouncement(c: ChannelAnnouncement) extends BadMessage
-  case class ChannelClosed(c: ChannelAnnouncement) extends BadMessage
+  object Gossip {
```

Comment:
I would even move that to its own separate file (in the router package), where we can start adding utility functions in the future. I would move `GossipOrigin` and some other classes that are currently directly defined in the router package (in `Router.scala`) inside this new `Gossip.scala` file.